commons/src/main/java/io/hyscale/commons/framework/event/publisher/EventPublisher.java

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright 2019 Pramati Prism, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.hyscale.commons.framework.event.publisher;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.config.ConfigurableBeanFactory;
+import org.springframework.context.ApplicationListener;
+import org.springframework.context.annotation.Scope;
+import org.springframework.context.event.SimpleApplicationEventMulticaster;
+import org.springframework.stereotype.Component;
+
+import io.hyscale.commons.framework.event.model.HyscaleEvent;
+import io.hyscale.commons.utils.HyscaleContextUtil;
+
+@Component
+@Scope(ConfigurableBeanFactory.SCOPE_SINGLETON)
+@SuppressWarnings("java:S3740")
+public class EventPublisher {
+    private static final Logger logger = LoggerFactory.getLogger(EventPublisher.class);
+
+    @Autowired
+    private SimpleApplicationEventMulticaster multicaster;
+
+    public void publishEvent(HyscaleEvent event) {
+        logger.debug("Publishing event: {}", event);
+        multicaster.multicastEvent(event);
+    }
+
+    public void removeListener(ApplicationListener listener) {
+        multicaster.removeApplicationListener(listener);
```

Comment:
While removing Listener we need to remove with listener as well as with listener name. So Its better to keep put those two at one place. No need to trigger 2 separate methods in the caller.

Reply:
Can you elaborate on this?
The reason to keep two different implementations is because of the underlying different implementations of SimpleApplicationEventMulticaster